### ISSUE TITLE
Performance/HashEachMethods report another block

### ISSUE THREAD

**Issue by znz** [bbatsov/rubocop#4881]
## Expected behavior

Do not detect `Performance/HashEachMethods` for block used key and value.
And detect `Lint/UnusedBlockArgument` only.

## Actual behavior

Detect `Performance/HashEachMethods` and `Lint/UnusedBlockArgument`.

## Steps to reproduce the problem

```
% cat r.rb
# frozen_string_literal: true

@h.each do |key, value|
  p(key * -value)
end
@h.sum { |key, value| -value }
% rubocop -D r.rb
Inspecting 1 file
W

Offenses:

r.rb:3:4: C: Performance/HashEachMethods: Use each_value instead of each.
@h.each do |key, value|
   ^^^^
r.rb:6:11: W: Lint/UnusedBlockArgument: Unused block argument - key. If it's necessary, use _ or _key as an argument name to indicate that it won't be used.
@h.sum { |key, value| -value }
          ^^^

1 file inspected, 2 offenses detected
```

## RuboCop version

```
% rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.5 x86_64-darwin16)
```
